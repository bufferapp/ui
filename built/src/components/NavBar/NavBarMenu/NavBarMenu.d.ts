import React from 'react';
import PropTypes from 'prop-types';
export declare function getUserAvatar(user: any): any;
declare const NavBarMenu: {
    (props: any): React.JSX.Element;
    propTypes: {
        user: PropTypes.Validator<PropTypes.InferProps<{
            name: PropTypes.Validator<string>;
            email: PropTypes.Validator<string>;
        }>>;
        isImpersonation: PropTypes.Requireable<boolean>;
    };
    defaultProps: {
        isImpersonation: boolean;
    };
};
export default NavBarMenu;
//# sourceMappingURL=NavBarMenu.d.ts.map