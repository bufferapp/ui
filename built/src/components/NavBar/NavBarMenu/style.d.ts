export declare const NavBarStyled: any;
export declare const NavBarUser: any;
export declare const NavBarImpersonating: any;
export declare const NavBarAvatar: any;
export declare const NavBarName: any;
export declare const NavBarEmail: any;
export declare const NavBarChavron: any;
//# sourceMappingURL=style.d.ts.map