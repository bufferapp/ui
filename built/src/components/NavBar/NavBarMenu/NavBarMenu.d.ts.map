{"version":3,"file":"NavBarMenu.d.ts","sourceRoot":"","sources":["../../../../../src/components/NavBar/NavBarMenu/NavBarMenu.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAalC,wBAAgB,aAAa,CAAC,IAAI,KAAA,OAMjC;AAKD,QAAA,MAAM,UAAU;;;;;;;;;;;;CAuBf,CAAA;AAeD,eAAe,UAAU,CAAA"}