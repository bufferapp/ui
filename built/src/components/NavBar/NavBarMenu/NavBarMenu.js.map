{"version":3,"file":"NavBarMenu.js","sourceRoot":"","sources":["../../../../../src/components/NavBar/NavBarMenu/NavBarMenu.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAClC,OAAO,EAAE,WAAW,EAAE,MAAM,YAAY,CAAA;AACxC,OAAO,EACL,YAAY,EACZ,WAAW,EACX,UAAU,EACV,UAAU,EACV,YAAY,EACZ,aAAa,EACb,mBAAmB,GACpB,MAAM,SAAS,CAAA;AAGhB,MAAM,UAAU,aAAa,CAAC,IAAI;IAChC,IAAI,IAAI,CAAC,MAAM,EAAE;QACf,OAAO,IAAI,CAAC,MAAM,CAAA;KACnB;IAED,OAAO,uDAAuD,CAAA;AAChE,CAAC;AAKD,MAAM,UAAU,GAAG,CAAC,KAAK,EAAE,EAAE;IAC3B,MAAM,EAAE,IAAI,EAAE,eAAe,EAAE,GAAG,KAAK,CAAA;IAEvC,OAAO,CACL,oBAAC,YAAY,oBAAK,KAAK;QACrB,oBAAC,UAAU;YACR,eAAe,IAAI,CAClB,oBAAC,mBAAmB,IAClB,IAAI,EAAE,IAAI,gBACC,8BAA8B,oBAGrB,CACvB;YACD,oBAAC,UAAU,QAAE,IAAI,CAAC,IAAI,CAAc;YACpC,oBAAC,WAAW,QAAE,IAAI,CAAC,KAAK,CAAe,CAC5B;QACb,oBAAC,YAAY,IAAC,MAAM,EAAE,aAAa,CAAC,IAAI,CAAC,GAAI;QAC7C,oBAAC,aAAa;YACZ,oBAAC,WAAW,IAAC,KAAK,EAAC,WAAW,EAAC,IAAI,EAAC,OAAO,GAAG,CAChC,CACH,CAChB,CAAA;AACH,CAAC,CAAA;AAED,UAAU,CAAC,SAAS,GAAG;IAErB,IAAI,EAAE,SAAS,CAAC,KAAK,CAAC;QACpB,IAAI,EAAE,SAAS,CAAC,MAAM,CAAC,UAAU;QACjC,KAAK,EAAE,SAAS,CAAC,MAAM,CAAC,UAAU;KACnC,CAAC,CAAC,UAAU;IACb,eAAe,EAAE,SAAS,CAAC,IAAI;CAChC,CAAA;AAED,UAAU,CAAC,YAAY,GAAG;IACxB,eAAe,EAAE,KAAK;CACvB,CAAA;AAED,eAAe,UAAU,CAAA","sourcesContent":["import React from 'react'\nimport PropTypes from 'prop-types'\nimport { ChevronDown } from '../../Icon'\nimport {\n  NavBarStyled,\n  NavBarEmail,\n  NavBarName,\n  NavBarUser,\n  NavBarAvatar,\n  NavBarChavron,\n  NavBarImpersonating,\n} from './style'\n\n// @ts-expect-error TS(7006) FIXME: Parameter 'user' implicitly has an 'any' type.\nexport function getUserAvatar(user) {\n  if (user.avatar) {\n    return user.avatar\n  }\n\n  return 'https://s3.amazonaws.com/buffer-ui/Default+Avatar.png'\n}\n\n/** NavBar Menu component used by the Select component to show a custom User name and avatar\n *  button */\n// @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.\nconst NavBarMenu = (props) => {\n  const { user, isImpersonation } = props\n\n  return (\n    <NavBarStyled {...props}>\n      <NavBarUser>\n        {isImpersonation && (\n          <NavBarImpersonating\n            user={user}\n            aria-label=\"You are impersonating a user\"\n          >\n            Impersonating\n          </NavBarImpersonating>\n        )}\n        <NavBarName>{user.name}</NavBarName>\n        <NavBarEmail>{user.email}</NavBarEmail>\n      </NavBarUser>\n      <NavBarAvatar avatar={getUserAvatar(user)} />\n      <NavBarChavron>\n        <ChevronDown color=\"grayLight\" size=\"large\" />\n      </NavBarChavron>\n    </NavBarStyled>\n  )\n}\n\nNavBarMenu.propTypes = {\n  /** User Name and Email to be shown in the NavBar */\n  user: PropTypes.shape({\n    name: PropTypes.string.isRequired,\n    email: PropTypes.string.isRequired,\n  }).isRequired,\n  isImpersonation: PropTypes.bool,\n}\n\nNavBarMenu.defaultProps = {\n  isImpersonation: false,\n}\n\nexport default NavBarMenu\n"]}