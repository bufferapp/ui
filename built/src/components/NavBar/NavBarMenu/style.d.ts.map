{"version":3,"file":"style.d.ts","sourceRoot":"","sources":["../../../../../src/components/NavBar/NavBarMenu/style.ts"],"names":[],"mappings":"AAYA,eAAO,MAAM,YAAY,KAYxB,CAAA;AAED,eAAO,MAAM,UAAU,KAKtB,CAAA;AAED,eAAO,MAAM,mBAAmB,KAY/B,CAAA;AAED,eAAO,MAAM,YAAY,KAOxB,CAAA;AAED,eAAO,MAAM,UAAU,KAStB,CAAA;AAED,eAAO,MAAM,WAAW,KAQvB,CAAA;AAED,eAAO,MAAM,aAAa,KAczB,CAAA"}