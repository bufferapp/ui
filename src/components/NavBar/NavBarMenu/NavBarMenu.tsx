import React from 'react'
import PropTypes from 'prop-types'
import { ChevronDown } from '../../Icon'
import {
  NavBarStyled,
  NavBarEmail,
  NavBarName,
  NavBarUser,
  NavBarAvatar,
  NavBarChavron,
  NavBarImpersonating,
} from './style'

// @ts-expect-error TS(7006) FIXME: Parameter 'user' implicitly has an 'any' type.
export function getUserAvatar(user) {
  if (user.avatar) {
    return user.avatar
  }

  return 'https://s3.amazonaws.com/buffer-ui/Default+Avatar.png'
}

/** NavBar Menu component used by the Select component to show a custom User name and avatar
 *  button */
// @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
const NavBarMenu = (props) => {
  const { user, isImpersonation } = props

  return (
    <NavBarStyled {...props}>
      <NavBarUser>
        {isImpersonation && (
          <NavBarImpersonating
            user={user}
            aria-label="You are impersonating a user"
          >
            Impersonating
          </NavBarImpersonating>
        )}
        <NavBarName>{user.name}</NavBarName>
        <NavBarEmail>{user.email}</NavBarEmail>
      </NavBarUser>
      <NavBarAvatar avatar={getUserAvatar(user)} />
      <NavBarChavron>
        <ChevronDown color="grayLight" size="large" />
      </NavBarChavron>
    </NavBarStyled>
  )
}

NavBarMenu.propTypes = {
  /** User Name and Email to be shown in the NavBar */
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
  isImpersonation: PropTypes.bool,
}

NavBarMenu.defaultProps = {
  isImpersonation: false,
}

export default NavBarMenu
