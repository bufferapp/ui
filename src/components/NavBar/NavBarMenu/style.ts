import styled from 'styled-components'
import {
  gray,
  grayDark,
  grayDarker,
  grayLight,
  grayLighter,
  white,
} from '../../style/colors'

import { fontSize, fontWeightMedium, fontFamily } from '../../style/fonts'

export const NavBarStyled = styled.a`
  align-items: center;
  cursor: pointer;
  display: flex;
  height: 100%;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  z-index: 2;
  &:hover {
    background-color: ${grayLighter};
  }
`

export const NavBarUser = styled.div`
  display: flex;
  align-items: end;
  flex-direction: column;
  padding-right: 16px;
`

export const NavBarImpersonating = styled.div`
  font-family: ${fontFamily};
  background: ${grayDarker};
  font-size: 12px;
  font-weight: bold;
  padding: 2px 8px;
  color: ${white};
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  margin-top: ${(props): string => (props.user.name ? '-8px' : '-21px')};
  margin-bottom: ${(props): string => (props.user.name ? '0px' : '5px')};
  text-transform: uppercase;
`

export const NavBarAvatar = styled.div`
  border-radius: 100px;
  width: 40px;
  height: 40px;
  background: ${gray};
  background-size: cover;
  background-image: url(${(props): string => props.avatar});
`

export const NavBarName = styled.div`
  color: ${grayDark};
  font-family: ${fontFamily};
  font-size: 16px;
  font-weight: ${fontWeightMedium};
  max-width: 224px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

export const NavBarEmail = styled.div`
  color: ${gray};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  max-width: 224px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

export const NavBarChavron = styled.div`
  align-items: center;
  background: ${white};
  border-radius: 4px;
  box-shadow: 0 0 0 2px ${grayLight};
  box-sizing: border-box;
  color: ${grayDarker};
  cursor: pointer;
  display: flex;
  height: 16px;
  position: absolute;
  right: 8px;
  width: 16px;
  justify-content: center;
`
