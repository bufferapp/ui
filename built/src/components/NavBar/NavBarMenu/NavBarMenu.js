import React from 'react';
import PropTypes from 'prop-types';
import { ChevronDown } from '../../Icon';
import { NavBarStyled, NavBarEmail, NavBarName, NavBarUser, NavBarAvatar, NavBarChavron, NavBarImpersonating, } from './style';
export function getUserAvatar(user) {
    if (user.avatar) {
        return user.avatar;
    }
    return 'https://s3.amazonaws.com/buffer-ui/Default+Avatar.png';
}
const NavBarMenu = (props) => {
    const { user, isImpersonation } = props;
    return (React.createElement(NavBarStyled, Object.assign({}, props),
        React.createElement(NavBarUser, null,
            isImpersonation && (React.createElement(NavBarImpersonating, { user: user, "aria-label": "You are impersonating a user" }, "Impersonating")),
            React.createElement(NavBarName, null, user.name),
            React.createElement(NavBarEmail, null, user.email)),
        React.createElement(NavBarAvatar, { avatar: getUserAvatar(user) }),
        React.createElement(NavBarChavron, null,
            React.createElement(ChevronDown, { color: "grayLight", size: "large" }))));
};
NavBarMenu.propTypes = {
    user: PropTypes.shape({
        name: PropTypes.string.isRequired,
        email: PropTypes.string.isRequired,
    }).isRequired,
    isImpersonation: PropTypes.bool,
};
NavBarMenu.defaultProps = {
    isImpersonation: false,
};
export default NavBarMenu;
//# sourceMappingURL=NavBarMenu.js.map