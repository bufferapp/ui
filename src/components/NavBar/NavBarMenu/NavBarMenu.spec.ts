// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
import NavBarMenu, { getUserAvatar } from './NavBarMenu'
import 'jest-styled-components'

describe('Avatar', () => {
  it('should return existing avatar URL', () => {
    expect(getUserAvatar({ avatar: 'foo' })).toBe(`foo`)
  })
  it('should return default', () => {
    const user = { email: 'foo@bar.com' }
    expect(getUserAvatar(user)).toBe(
      `https://s3.amazonaws.com/buffer-ui/Default+Avatar.png`,
    )
  })
})

describe('NavBarMenu methods', () => {})

snap(NavBarMenu, './NavBarMenu.tsx')
