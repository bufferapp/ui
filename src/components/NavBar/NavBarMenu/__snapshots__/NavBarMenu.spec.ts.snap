// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > NavBarMenu Matches snapshot when boolean prop "isImpersonation" is set to: "false" 1`] = `
<styled.a
  isImpersonation={false}
  user={
    Object {
      "email": "jest-auto-snapshots String Fixture",
      "name": "jest-auto-snapshots String Fixture",
    }
  }
>
  <styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
  </styled.div>
  <styled.div
    avatar="https://s3.amazonaws.com/buffer-ui/Default+Avatar.png"
  />
  <styled.div>
    <ChevronDownIcon
      color="grayLight"
      size="large"
    />
  </styled.div>
</styled.a>
`;

exports[`jest-auto-snapshots > NavBarMenu Matches snapshot when passed all props 1`] = `
<styled.a
  isImpersonation={true}
  user={
    Object {
      "email": "jest-auto-snapshots String Fixture",
      "name": "jest-auto-snapshots String Fixture",
    }
  }
>
  <styled.div>
    <styled.div
      aria-label="You are impersonating a user"
      user={
        Object {
          "email": "jest-auto-snapshots String Fixture",
          "name": "jest-auto-snapshots String Fixture",
        }
      }
    >
      Impersonating
    </styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
  </styled.div>
  <styled.div
    avatar="https://s3.amazonaws.com/buffer-ui/Default+Avatar.png"
  />
  <styled.div>
    <ChevronDownIcon
      color="grayLight"
      size="large"
    />
  </styled.div>
</styled.a>
`;

exports[`jest-auto-snapshots > NavBarMenu Matches snapshot when passed only required props 1`] = `
<styled.a
  isImpersonation={false}
  user={
    Object {
      "email": "jest-auto-snapshots String Fixture",
      "name": "jest-auto-snapshots String Fixture",
    }
  }
>
  <styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
    <styled.div>
      jest-auto-snapshots String Fixture
    </styled.div>
  </styled.div>
  <styled.div
    avatar="https://s3.amazonaws.com/buffer-ui/Default+Avatar.png"
  />
  <styled.div>
    <ChevronDownIcon
      color="grayLight"
      size="large"
    />
  </styled.div>
</styled.a>
`;
